### PR TITLE
[tensorflow] [build] [test] Updating TF binaries for 2.3.1 CUDA10.2

### PR DESCRIPTION
**Pull request by satishpasumarthi**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
- Dockefiles for CPU and GPU (cuda10.2)
- No change in estimator, TFS binaries
*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/2.3.1/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ARG PYTHON_VERSION=3.7.7
+ARG OPENSSL_VERSION=1.1.1g
+
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.3_aws/20201002-134350/cpu/py37/tensorflow_cpu-2.3.1-cp37-cp37m-manylinux2010_x86_64.whl
+
+ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/estimator/r2.3_aws/20201002-134350/tensorflow_estimator-2.3.0-py2.py3-none-any.whl
+
+# The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
+# the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (NihalHarish) [inline tensorflow/training/docker/2.3.1/py3/Dockerfile.cpu:37]: Bump to 0.9.4

File: tensorflow/training/docker/2.3.1/py3/cu102/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+ARG PYTHON_VERSION=3.7.7
+ARG OPENSSL_VERSION=1.1.1g
+
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.3_aws/20201002-134350/gpu/py37/tensorflow_gpu-2.3.1-cp37-cp37m-manylinux2010_x86_64.whl
+
+ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/estimator/r2.3_aws/20201002-134350/tensorflow_estimator-2.3.0-py2.py3-none-any.whl
+
+# The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
+# the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (NihalHarish) [inline tensorflow/training/docker/2.3.1/py3/cu102/Dockerfile.gpu:36]: 0.9.4

Comment (satishpasumarthi) [reply tensorflow/training/docker/2.3.1/py3/cu102/Dockerfile.gpu]: done

File: tensorflow/training/docker/2.3.1/py3/Dockerfile.cpu (hunk 1 above)
Comment (satishpasumarthi) [reply tensorflow/training/docker/2.3.1/py3/Dockerfile.cpu]: done

**Review by arjkesh — APPROVED** [review]

**Review by saimidu — APPROVED** [review]

**Pull request merged**